### PR TITLE
Fix onnx importer to treat Constant values as static

### PR DESCRIPTION
**Pull request by daveliddell**
Fixes  https://github.com/llvm/torch-mlir/issues/2764

In the case of OPT, there are ConstantOfShape ops whose input shape is not static (that is, an initializer), but rather comes from a Constant op.  The importer can't handle such non-static input shapes.

The fix here is to create initializers for a subset of Constant ops (ones with "value" attributes), so that their outputs can be used statically.  Additionally, there was no case for creating a splat of int64, so I added that as well.

### CODE REVIEW

File: python/torch_mlir/extras/onnx_importer.py [hunk 1]
```diff
@@ -276,10 +276,13 @@ def import_node(self, node: onnx.NodeProto):
         with InsertionPoint(self._b), Location.name(node.name):
             op_type = node.op_type
             # Handle special op types that materialize to non-op IR constructs.
+            # Handlers return True if the op was handled, else this function
+            # should process it as a general node.
             special_key = f"_handle_node_{op_type}"
             if hasattr(self, special_key):
-                getattr(self, special_key)(node)
-                return
+                wasHandled = getattr(self, special_key)(node)
```
Comment (stellaraccident) [inline python/torch_mlir/extras/onnx_importer.py:283]: Nit: Use snake_case

File: python/torch_mlir/extras/onnx_importer.py [hunk 2]
```diff
@@ -605,9 +633,10 @@ def tensor_proto_to_attr(self, tp: onnx.TensorProto) -> Attribute:
 }
 
 
-def _get_attr(node: onnx.NodeProto, attr_name: str) -> onnx.AttributeProto:
+def _get_attr(node: onnx.NodeProto, attr_name: str, isRequired: bool = True) -> onnx.AttributeProto:
     for attr in node.attribute:
         if attr.name == attr_name:
             return attr
-    else:
+    if isRequired:
```
Comment (stellaraccident) [inline python/torch_mlir/extras/onnx_importer.py:640]: Nit: snake_case

File: python/torch_mlir/extras/onnx_importer.py [hunk 3]
```diff
@@ -605,9 +633,10 @@ def tensor_proto_to_attr(self, tp: onnx.TensorProto) -> Attribute:
...
         raise OnnxImportError(f"Required attribute {attr_name} not found in {node}")
+    return None
```
Comment (stellaraccident) [inline python/torch_mlir/extras/onnx_importer.py:642]: Nit: newline at end

**Review by stellaraccident — APPROVED** [review]
A few small nits then good to go. Thanks!

**Comment by stellaraccident** [comment]
Interesting that this kind of thing doesn't seem to be in the onnx test suite... (would have caused an xfail to start passing)

File: python/torch_mlir/extras/onnx_importer.py (hunk 1 above)
Comment (daveliddell) [reply python/torch_mlir/extras/onnx_importer.py]: D'oh!  Why isn't my editor doing my thinking for me?!  Fixed

File: python/torch_mlir/extras/onnx_importer.py (hunk 2 above)
Comment (daveliddell) [reply python/torch_mlir/extras/onnx_importer.py]: Fixed.  I thought we didn't like snakes here

**Pull request merged**